festim/heat_transfer_problem.py

```diff
@@ -21,21 +12,11 @@ def __init__(
         exports=None,
         settings=None,
     ) -> None:
-        self.mesh = mesh
-        self.subdomains = subdomains or []
+        super().__init__(
+            mesh, sources, exports, subdomains, boundary_conditions, settings
```

Comment:
should we use the keywords here `mesh=mesh, sources=sources` etc

This would avoid errors when/if we change the order of the arguments in `Problem` one day.

Same comment for `HTransportProblem`